diff --git a/ar-lib b/ar-lib
old mode 100755
new mode 100644
diff --git a/c++tools/configure b/c++tools/configure
old mode 100755
new mode 100644
diff --git a/compile b/compile
old mode 100755
new mode 100644
diff --git a/config.guess b/config.guess
old mode 100755
new mode 100644
diff --git a/config.rpath b/config.rpath
old mode 100755
new mode 100644
diff --git a/config.sub b/config.sub
old mode 100755
new mode 100644
index 38f3d037a78..6af9d2083cd
--- a/config.sub
+++ b/config.sub
@@ -1749,7 +1749,7 @@ case $os in
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | goose* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff --git a/configure b/configure
old mode 100755
new mode 100644
diff --git a/contrib/analyze_brprob.py b/contrib/analyze_brprob.py
old mode 100755
new mode 100644
diff --git a/contrib/analyze_brprob_spec.py b/contrib/analyze_brprob_spec.py
old mode 100755
new mode 100644
diff --git a/contrib/bench-stringop b/contrib/bench-stringop
old mode 100755
new mode 100644
diff --git a/contrib/check-MAINTAINERS.py b/contrib/check-MAINTAINERS.py
old mode 100755
new mode 100644
diff --git a/contrib/check-internal-format-escaping.py b/contrib/check-internal-format-escaping.py
old mode 100755
new mode 100644
diff --git a/contrib/check-params-in-docs.py b/contrib/check-params-in-docs.py
old mode 100755
new mode 100644
diff --git a/contrib/check_GNU_style.py b/contrib/check_GNU_style.py
old mode 100755
new mode 100644
diff --git a/contrib/check_GNU_style.sh b/contrib/check_GNU_style.sh
old mode 100755
new mode 100644
diff --git a/contrib/check_GNU_style_lib.py b/contrib/check_GNU_style_lib.py
old mode 100755
new mode 100644
diff --git a/contrib/check_warning_flags.sh b/contrib/check_warning_flags.sh
old mode 100755
new mode 100644
diff --git a/contrib/compare-debug b/contrib/compare-debug
old mode 100755
new mode 100644
diff --git a/contrib/compare-lto b/contrib/compare-lto
old mode 100755
new mode 100644
diff --git a/contrib/compareSumTests3 b/contrib/compareSumTests3
old mode 100755
new mode 100644
diff --git a/contrib/compare_tests b/contrib/compare_tests
old mode 100755
new mode 100644
diff --git a/contrib/compare_two_ftime_report_sets b/contrib/compare_two_ftime_report_sets
old mode 100755
new mode 100644
diff --git a/contrib/dg-cmp-results.sh b/contrib/dg-cmp-results.sh
old mode 100755
new mode 100644
diff --git a/contrib/dg-extract-results.sh b/contrib/dg-extract-results.sh
old mode 100755
new mode 100644
diff --git a/contrib/dg-out-generator.pl b/contrib/dg-out-generator.pl
old mode 100755
new mode 100644
diff --git a/contrib/download_prerequisites b/contrib/download_prerequisites
old mode 100755
new mode 100644
diff --git a/contrib/filter-clang-warnings.py b/contrib/filter-clang-warnings.py
old mode 100755
new mode 100644
diff --git a/contrib/filter_gcc_for_doxygen b/contrib/filter_gcc_for_doxygen
old mode 100755
new mode 100644
diff --git a/contrib/filter_knr2ansi.pl b/contrib/filter_knr2ansi.pl
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/git_check_commit.py b/contrib/gcc-changelog/git_check_commit.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/git_commit.py b/contrib/gcc-changelog/git_commit.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/git_email.py b/contrib/gcc-changelog/git_email.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/git_repository.py b/contrib/gcc-changelog/git_repository.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/git_update_version.py b/contrib/gcc-changelog/git_update_version.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-changelog/test_email.py b/contrib/gcc-changelog/test_email.py
old mode 100755
new mode 100644
diff --git a/contrib/gcc-git-customization.sh b/contrib/gcc-git-customization.sh
old mode 100755
new mode 100644
diff --git a/contrib/gcc_build b/contrib/gcc_build
old mode 100755
new mode 100644
diff --git a/contrib/gcc_update b/contrib/gcc_update
old mode 100755
new mode 100644
diff --git a/contrib/gen_autofdo_event.py b/contrib/gen_autofdo_event.py
old mode 100755
new mode 100644
diff --git a/contrib/gennews b/contrib/gennews
old mode 100755
new mode 100644
diff --git a/contrib/git-add-user-branch.sh b/contrib/git-add-user-branch.sh
old mode 100755
new mode 100644
diff --git a/contrib/git-add-vendor-branch.sh b/contrib/git-add-vendor-branch.sh
old mode 100755
new mode 100644
diff --git a/contrib/git-backport.py b/contrib/git-backport.py
old mode 100755
new mode 100644
diff --git a/contrib/git-commit-mklog.py b/contrib/git-commit-mklog.py
old mode 100755
new mode 100644
diff --git a/contrib/git-descr.sh b/contrib/git-descr.sh
old mode 100755
new mode 100644
diff --git a/contrib/git-fetch-vendor.sh b/contrib/git-fetch-vendor.sh
old mode 100755
new mode 100644
diff --git a/contrib/git-fix-changelog.py b/contrib/git-fix-changelog.py
old mode 100755
new mode 100644
diff --git a/contrib/git-undescr.sh b/contrib/git-undescr.sh
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/count-headers b/contrib/header-tools/count-headers
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/gcc-order-headers b/contrib/header-tools/gcc-order-headers
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/graph-header-logs b/contrib/header-tools/graph-header-logs
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/graph-include-web b/contrib/header-tools/graph-include-web
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/headerutils.py b/contrib/header-tools/headerutils.py
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/included-by b/contrib/header-tools/included-by
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/reduce-headers b/contrib/header-tools/reduce-headers
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/replace-header b/contrib/header-tools/replace-header
old mode 100755
new mode 100644
diff --git a/contrib/header-tools/show-headers b/contrib/header-tools/show-headers
old mode 100755
new mode 100644
diff --git a/contrib/index-prop b/contrib/index-prop
old mode 100755
new mode 100644
diff --git a/contrib/legacy/mklog b/contrib/legacy/mklog
old mode 100755
new mode 100644
diff --git a/contrib/legacy/mklog.pl b/contrib/legacy/mklog.pl
old mode 100755
new mode 100644
diff --git a/contrib/mark_spam.py b/contrib/mark_spam.py
old mode 100755
new mode 100644
diff --git a/contrib/mklog.py b/contrib/mklog.py
old mode 100755
new mode 100644
diff --git a/contrib/patch_tester.sh b/contrib/patch_tester.sh
old mode 100755
new mode 100644
diff --git a/contrib/prepare-commit-msg b/contrib/prepare-commit-msg
old mode 100755
new mode 100644
diff --git a/contrib/prepare_patch.sh b/contrib/prepare_patch.sh
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-build-full b/contrib/reghunt/bin/gcc-build-full
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-build-simple b/contrib/reghunt/bin/gcc-build-simple
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-cleanup b/contrib/reghunt/bin/gcc-cleanup
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-checkfail b/contrib/reghunt/bin/gcc-svn-checkfail
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-ids b/contrib/reghunt/bin/gcc-svn-ids
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-patchlist b/contrib/reghunt/bin/gcc-svn-patchlist
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-recordfail b/contrib/reghunt/bin/gcc-svn-recordfail
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-report b/contrib/reghunt/bin/gcc-svn-report
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-update b/contrib/reghunt/bin/gcc-svn-update
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-svn-update-fix b/contrib/reghunt/bin/gcc-svn-update-fix
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-accepts-invalid b/contrib/reghunt/bin/gcc-test-accepts-invalid
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-bogus-warning b/contrib/reghunt/bin/gcc-test-bogus-warning
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-compiler-hangs b/contrib/reghunt/bin/gcc-test-compiler-hangs
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-ice-on-invalid-code b/contrib/reghunt/bin/gcc-test-ice-on-invalid-code
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-ice-on-valid-code b/contrib/reghunt/bin/gcc-test-ice-on-valid-code
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-missing-warning b/contrib/reghunt/bin/gcc-test-missing-warning
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-rejects-valid b/contrib/reghunt/bin/gcc-test-rejects-valid
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-segfault-on-valid-code b/contrib/reghunt/bin/gcc-test-segfault-on-valid-code
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/gcc-test-wrong-code b/contrib/reghunt/bin/gcc-test-wrong-code
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/reg-hunt b/contrib/reghunt/bin/reg-hunt
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/reg-newmid b/contrib/reghunt/bin/reg-newmid
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/bin/reg-test b/contrib/reghunt/bin/reg-test
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/date_based/reg_periodic b/contrib/reghunt/date_based/reg_periodic
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/date_based/reg_search b/contrib/reghunt/date_based/reg_search
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/date_based/reg_test_template b/contrib/reghunt/date_based/reg_test_template
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/examples/29106.test b/contrib/reghunt/examples/29106.test
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/examples/30643.test b/contrib/reghunt/examples/30643.test
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/examples/reg-watch b/contrib/reghunt/examples/reg-watch
old mode 100755
new mode 100644
diff --git a/contrib/reghunt/examples/testall b/contrib/reghunt/examples/testall
old mode 100755
new mode 100644
diff --git a/contrib/regression/btest-gcc.sh b/contrib/regression/btest-gcc.sh
old mode 100755
new mode 100644
diff --git a/contrib/regression/mkindex.pl b/contrib/regression/mkindex.pl
old mode 100755
new mode 100644
diff --git a/contrib/regression/objs-gcc.sh b/contrib/regression/objs-gcc.sh
old mode 100755
new mode 100644
diff --git a/contrib/repro_fail b/contrib/repro_fail
old mode 100755
new mode 100644
diff --git a/contrib/test_installed b/contrib/test_installed
old mode 100755
new mode 100644
diff --git a/contrib/test_mklog.py b/contrib/test_mklog.py
old mode 100755
new mode 100644
diff --git a/contrib/test_recheck b/contrib/test_recheck
old mode 100755
new mode 100644
diff --git a/contrib/test_summary b/contrib/test_summary
old mode 100755
new mode 100644
diff --git a/contrib/testsuite-management/validate_failures.py b/contrib/testsuite-management/validate_failures.py
old mode 100755
new mode 100644
diff --git a/contrib/texi2pod.pl b/contrib/texi2pod.pl
old mode 100755
new mode 100644
diff --git a/contrib/unicode/from_glibc/utf8_gen.py b/contrib/unicode/from_glibc/utf8_gen.py
old mode 100755
new mode 100644
diff --git a/contrib/unicode/gen_wcwidth.py b/contrib/unicode/gen_wcwidth.py
old mode 100755
new mode 100644
diff --git a/contrib/unicode/utf8-dump.py b/contrib/unicode/utf8-dump.py
old mode 100755
new mode 100644
diff --git a/contrib/uninclude b/contrib/uninclude
old mode 100755
new mode 100644
diff --git a/contrib/unused_functions.py b/contrib/unused_functions.py
old mode 100755
new mode 100644
diff --git a/contrib/update-copyright.py b/contrib/update-copyright.py
old mode 100755
new mode 100644
diff --git a/contrib/warn_summary b/contrib/warn_summary
old mode 100755
new mode 100644
diff --git a/depcomp b/depcomp
old mode 100755
new mode 100644
diff --git a/fixincludes/configure b/fixincludes/configure
old mode 100755
new mode 100644
diff --git a/fixincludes/fixinc.in b/fixincludes/fixinc.in
old mode 100755
new mode 100644
diff --git a/fixincludes/genfixes b/fixincludes/genfixes
old mode 100755
new mode 100644
diff --git a/fixincludes/mkfixinc.sh b/fixincludes/mkfixinc.sh
old mode 100755
new mode 100644
index df90720b716..e99f9c330b8
--- a/fixincludes/mkfixinc.sh
+++ b/fixincludes/mkfixinc.sh
@@ -11,6 +11,8 @@ target=fixinc.sh
 
 # Check for special fix rules for particular targets
 case $machine in
+    *-goose* | \
+    *-*-goose* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
     x86_64-*-mingw32* | \
diff --git a/gcc/config.gcc b/gcc/config.gcc
index 648b3dc2110..5e45b18e0e4 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -690,6 +690,12 @@ x86_cpus="generic intel"
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-goose*)
+  gas=yes
+  gnu_ld=yes
+  default_use_xca_atexit=yes
+  use_gcc_stdint=provide
+  ;;
 *-*-darwin*)
   tmake_file="t-darwin "
   tm_file="${tm_file} darwin.h"
@@ -1126,6 +1132,12 @@ case ${target} in
 esac
 
 case ${target} in
+i[34567]86-*-goose*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h goose.h"
+	;;
+x86_64-*-goose*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h i386/x86-64.h goose.h"
+	;;
 aarch64*-*-elf | aarch64*-*-fuchsia* | aarch64*-*-rtems*)
 	tm_file="${tm_file} elfos.h newlib-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-errata.h aarch64/aarch64-elf-raw.h"
diff --git a/gcc/config/i386/gcc-auto-profile b/gcc/config/i386/gcc-auto-profile
old mode 100755
new mode 100644
diff --git a/gcc/config/loongarch/genopts/genstr.sh b/gcc/config/loongarch/genopts/genstr.sh
old mode 100755
new mode 100644
diff --git a/gcc/config/m68k/genopt.sh b/gcc/config/m68k/genopt.sh
old mode 100755
new mode 100644
diff --git a/gcc/config/mips/genopt.sh b/gcc/config/mips/genopt.sh
old mode 100755
new mode 100644
diff --git a/gcc/config/nvptx/gen-multilib-matches.sh b/gcc/config/nvptx/gen-multilib-matches.sh
old mode 100755
new mode 100644
diff --git a/gcc/config/riscv/arch-canonicalize b/gcc/config/riscv/arch-canonicalize
old mode 100755
new mode 100644
diff --git a/gcc/config/riscv/multilib-generator b/gcc/config/riscv/multilib-generator
old mode 100755
new mode 100644
diff --git a/gcc/config/rs6000/genfusion.pl b/gcc/config/rs6000/genfusion.pl
old mode 100755
new mode 100644
diff --git a/gcc/config/rs6000/genopt.sh b/gcc/config/rs6000/genopt.sh
old mode 100755
new mode 100644
diff --git a/gcc/configure b/gcc/configure
old mode 100755
new mode 100644
diff --git a/gcc/doc/install.texi2html b/gcc/doc/install.texi2html
old mode 100755
new mode 100644
diff --git a/gcc/m2/configure b/gcc/m2/configure
old mode 100755
new mode 100644
diff --git a/gcc/m2/gm2-libs/config-host b/gcc/m2/gm2-libs/config-host
old mode 100755
new mode 100644
diff --git a/gcc/m2/images/LICENSE.IMG b/gcc/m2/images/LICENSE.IMG
old mode 100755
new mode 100644
diff --git a/gcc/m2/images/gnu.eps b/gcc/m2/images/gnu.eps
old mode 100755
new mode 100644
diff --git a/gcc/m2/images/gnupng b/gcc/m2/images/gnupng
old mode 100755
new mode 100644
diff --git a/gcc/m2/tools-src/calcpath b/gcc/m2/tools-src/calcpath
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/ada/acats/run_acats.sh b/gcc/testsuite/ada/acats/run_acats.sh
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/ada/acats/run_all.sh b/gcc/testsuite/ada/acats/run_all.sh
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/g++.dg/contracts/backtrace_handler/prettytrace.sh b/gcc/testsuite/g++.dg/contracts/backtrace_handler/prettytrace.sh
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gcc.target/gcn/complex.c b/gcc/testsuite/gcc.target/gcn/complex.c
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gcc.target/i386/pr104371-2.c b/gcc/testsuite/gcc.target/i386/pr104371-2.c
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gcc.target/i386/vperm.pl b/gcc/testsuite/gcc.target/i386/vperm.pl
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gfortran.dg/dec_union_12.f90 b/gcc/testsuite/gfortran.dg/dec_union_12.f90
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gm2/iso/extended-opaque/pass/iso-extended-opaque-pass.exp b/gcc/testsuite/gm2/iso/extended-opaque/pass/iso-extended-opaque-pass.exp
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gm2/link/nondefaultext/pass/link-nondefaultext-pass.exp b/gcc/testsuite/gm2/link/nondefaultext/pass/link-nondefaultext-pass.exp
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/gm2/projects/iso/small/run/pass/iso-extended-opaque-run-pass.exp b/gcc/testsuite/gm2/projects/iso/small/run/pass/iso-extended-opaque-run-pass.exp
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/rust/compile/torture/shebang.rs b/gcc/testsuite/rust/compile/torture/shebang.rs
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/rust/compile/torture/shebang_plus_attr.rs b/gcc/testsuite/rust/compile/torture/shebang_plus_attr.rs
old mode 100755
new mode 100644
diff --git a/gcc/testsuite/rust/compile/torture/shebang_plus_attr2.rs b/gcc/testsuite/rust/compile/torture/shebang_plus_attr2.rs
old mode 100755
new mode 100644
diff --git a/gnattools/configure b/gnattools/configure
old mode 100755
new mode 100644
diff --git a/gotools/configure b/gotools/configure
old mode 100755
new mode 100644
diff --git a/install-sh b/install-sh
old mode 100755
new mode 100644
diff --git a/intl/configure b/intl/configure
old mode 100755
new mode 100644
diff --git a/libada/configure b/libada/configure
old mode 100755
new mode 100644
diff --git a/libatomic/configure b/libatomic/configure
old mode 100755
new mode 100644
diff --git a/libbacktrace/allocfail.sh b/libbacktrace/allocfail.sh
old mode 100755
new mode 100644
diff --git a/libbacktrace/configure b/libbacktrace/configure
old mode 100755
new mode 100644
diff --git a/libcc1/configure b/libcc1/configure
old mode 100755
new mode 100644
diff --git a/libcody/configure b/libcody/configure
old mode 100755
new mode 100644
diff --git a/libcpp/configure b/libcpp/configure
old mode 100755
new mode 100644
diff --git a/libdecnumber/configure b/libdecnumber/configure
old mode 100755
new mode 100644
diff --git a/libffi/autogen.sh b/libffi/autogen.sh
old mode 100755
new mode 100644
diff --git a/libffi/configure b/libffi/configure
old mode 100755
new mode 100644
diff --git a/libffi/generate-darwin-source-and-headers.py b/libffi/generate-darwin-source-and-headers.py
old mode 100755
new mode 100644
diff --git a/libffi/mdate-sh b/libffi/mdate-sh
old mode 100755
new mode 100644
diff --git a/libffi/merge.sh b/libffi/merge.sh
old mode 100755
new mode 100644
diff --git a/libffi/msvcc.sh b/libffi/msvcc.sh
old mode 100755
new mode 100644
diff --git a/libgcc/config.host b/libgcc/config.host
index 9d7212028d0..70b239727c0 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -386,6 +386,14 @@ i[34567]86-*-cygwin* | x86_64-*-cygwin*)
 esac
 
 case ${host} in
+i[34567]86-*-goose*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="${tmake_file} i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
+x86_64-*-goose*)	
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="${tmake_file} i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 aarch64*-*-elf | aarch64*-*-rtems*)
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
 	extra_parts="$extra_parts crtfastmath.o"
diff --git a/libgcc/config/avr/libf7/f7renames.sh b/libgcc/config/avr/libf7/f7renames.sh
old mode 100755
new mode 100644
diff --git a/libgcc/config/avr/libf7/f7wraps.sh b/libgcc/config/avr/libf7/f7wraps.sh
old mode 100755
new mode 100644
diff --git a/libgcc/configure b/libgcc/configure
old mode 100755
new mode 100644
diff --git a/libgcc/mkheader.sh b/libgcc/mkheader.sh
old mode 100755
new mode 100644
diff --git a/libgfortran/configure b/libgfortran/configure
old mode 100755
new mode 100644
diff --git a/libgfortran/mk-kinds-h.sh b/libgfortran/mk-kinds-h.sh
old mode 100755
new mode 100644
diff --git a/libgfortran/mk-sik-inc.sh b/libgfortran/mk-sik-inc.sh
old mode 100755
new mode 100644
diff --git a/libgfortran/mk-srk-inc.sh b/libgfortran/mk-srk-inc.sh
old mode 100755
new mode 100644
diff --git a/libgm2/configure b/libgm2/configure
old mode 100755
new mode 100644
diff --git a/libgo/configure b/libgo/configure
old mode 100755
new mode 100644
diff --git a/libgo/go/cmd/go/mkalldocs.sh b/libgo/go/cmd/go/mkalldocs.sh
old mode 100755
new mode 100644
diff --git a/libgo/go/debug/dwarf/testdata/rnglistx.elf b/libgo/go/debug/dwarf/testdata/rnglistx.elf
old mode 100755
new mode 100644
diff --git a/libgo/go/debug/dwarf/testdata/typedef.elf b/libgo/go/debug/dwarf/testdata/typedef.elf
old mode 100755
new mode 100644
diff --git a/libgo/go/debug/elf/testdata/gcc-386-freebsd-exec b/libgo/go/debug/elf/testdata/gcc-386-freebsd-exec
old mode 100755
new mode 100644
diff --git a/libgo/go/debug/elf/testdata/gcc-amd64-linux-exec b/libgo/go/debug/elf/testdata/gcc-amd64-linux-exec
old mode 100755
new mode 100644
diff --git a/libgo/go/runtime/pprof/testdata/test32 b/libgo/go/runtime/pprof/testdata/test32
old mode 100755
new mode 100644
diff --git a/libgo/go/runtime/pprof/testdata/test32be b/libgo/go/runtime/pprof/testdata/test32be
old mode 100755
new mode 100644
diff --git a/libgo/go/runtime/pprof/testdata/test64 b/libgo/go/runtime/pprof/testdata/test64
old mode 100755
new mode 100644
diff --git a/libgo/go/runtime/pprof/testdata/test64be b/libgo/go/runtime/pprof/testdata/test64be
old mode 100755
new mode 100644
diff --git a/libgo/goarch.sh b/libgo/goarch.sh
old mode 100755
new mode 100644
diff --git a/libgo/match.sh b/libgo/match.sh
old mode 100755
new mode 100644
diff --git a/libgo/merge.sh b/libgo/merge.sh
old mode 100755
new mode 100644
diff --git a/libgo/mkrsysinfo.sh b/libgo/mkrsysinfo.sh
old mode 100755
new mode 100644
diff --git a/libgo/mkruntimeinc.sh b/libgo/mkruntimeinc.sh
old mode 100755
new mode 100644
diff --git a/libgo/mksysinfo.sh b/libgo/mksysinfo.sh
old mode 100755
new mode 100644
diff --git a/libgo/testsuite/gotest b/libgo/testsuite/gotest
old mode 100755
new mode 100644
diff --git a/libgomp/configure b/libgomp/configure
old mode 100755
new mode 100644
diff --git a/libgomp/testsuite/flock b/libgomp/testsuite/flock
old mode 100755
new mode 100644
diff --git a/libiberty/configure b/libiberty/configure
old mode 100755
new mode 100644
diff --git a/libitm/configure b/libitm/configure
old mode 100755
new mode 100644
diff --git a/libobjc/configure b/libobjc/configure
old mode 100755
new mode 100644
diff --git a/libphobos/configure b/libphobos/configure
old mode 100755
new mode 100644
diff --git a/libphobos/libdruntime/core/sys/posix/sys/ioccom.d b/libphobos/libdruntime/core/sys/posix/sys/ioccom.d
old mode 100755
new mode 100644
diff --git a/libphobos/libdruntime/core/sys/posix/sys/ttycom.d b/libphobos/libdruntime/core/sys/posix/sys/ttycom.d
old mode 100755
new mode 100644
diff --git a/libphobos/testsuite/testsuite_flags.in b/libphobos/testsuite/testsuite_flags.in
old mode 100755
new mode 100644
diff --git a/libquadmath/configure b/libquadmath/configure
old mode 100755
new mode 100644
diff --git a/libquadmath/update-quadmath.py b/libquadmath/update-quadmath.py
old mode 100755
new mode 100644
diff --git a/libsanitizer/configure b/libsanitizer/configure
old mode 100755
new mode 100644
diff --git a/libsanitizer/merge.sh b/libsanitizer/merge.sh
old mode 100755
new mode 100644
diff --git a/libssp/configure b/libssp/configure
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/check_compile b/libstdc++-v3/scripts/check_compile
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/check_performance b/libstdc++-v3/scripts/check_performance
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/check_simd b/libstdc++-v3/scripts/check_simd
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/create_testsuite_files b/libstdc++-v3/scripts/create_testsuite_files
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/extract_symvers.in b/libstdc++-v3/scripts/extract_symvers.in
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/make_graph.py b/libstdc++-v3/scripts/make_graph.py
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/scripts/testsuite_flags.in b/libstdc++-v3/scripts/testsuite_flags.in
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/testsuite/experimental/simd/driver.sh b/libstdc++-v3/testsuite/experimental/simd/driver.sh
old mode 100755
new mode 100644
diff --git a/libstdc++-v3/testsuite/experimental/simd/generate_makefile.sh b/libstdc++-v3/testsuite/experimental/simd/generate_makefile.sh
old mode 100755
new mode 100644
diff --git a/libtool-ldflags b/libtool-ldflags
old mode 100755
new mode 100644
diff --git a/libvtv/configure b/libvtv/configure
old mode 100755
new mode 100644
diff --git a/lto-plugin/configure b/lto-plugin/configure
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/branch_changer.py b/maintainer-scripts/branch_changer.py
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/bugzilla-close-candidate.py b/maintainer-scripts/bugzilla-close-candidate.py
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/gcc_release b/maintainer-scripts/gcc_release
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/generate_libstdcxx_web_docs b/maintainer-scripts/generate_libstdcxx_web_docs
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/maintainer-addresses b/maintainer-scripts/maintainer-addresses
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/update_version_git b/maintainer-scripts/update_version_git
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/update_web_docs_git b/maintainer-scripts/update_web_docs_git
old mode 100755
new mode 100644
diff --git a/maintainer-scripts/update_web_docs_libstdcxx_git b/maintainer-scripts/update_web_docs_libstdcxx_git
old mode 100755
new mode 100644
diff --git a/missing b/missing
old mode 100755
new mode 100644
diff --git a/mkdep b/mkdep
old mode 100755
new mode 100644
diff --git a/mkinstalldirs b/mkinstalldirs
old mode 100755
new mode 100644
diff --git a/move-if-change b/move-if-change
old mode 100755
new mode 100644
diff --git a/symlink-tree b/symlink-tree
old mode 100755
new mode 100644
diff --git a/test-driver b/test-driver
old mode 100755
new mode 100644
diff --git a/ylwrap b/ylwrap
old mode 100755
new mode 100644
diff --git a/zlib/configure b/zlib/configure
old mode 100755
new mode 100644
